app/Console/Commands/PilotPassportActivityUpdate.php

```diff
@@ -0,0 +1,142 @@
+            if (is_null($airport_id)) {
+                continue;
+            }
+            if (!PilotPassportLog::where('cid', $flight->cid)->where('airfield', $airport_id)->get()->isEmpty()) {
+                $this->info('- [Visited Limit] - airfield has already been logged');
+                continue;
+            }
+
+            $passport = new PilotPassportLog;
+            $passport->cid = $flight->cid;
+            $passport->airfield = $airport_id;
+            $passport->visited_on = date('Y-m-d H:i:s');
+            $passport->callsign = $flight->callsign;
+            $passport->aircraft_type = (is_null($flight->flight_plan)) ? '' : $flight->flight_plan->aircraft_faa;
+            $passport->save();
+
+            $pilot = RealopsPilot::find($flight->cid);
+            Mail::to($pilot->email)->send(new PilotPassportMail('visited_airfield', $pilot, $airport_id));
+            $this->checkPhaseComplete($pilot);
+        }
+    }
+
+    private function isFlightAtAirport($flight) {
+        $ppos = new LatLong($flight->latitude, $flight->longitude);
+        foreach ($this->airports as $airport) {
+            $this->info('- Checking ' . $airport->id);
+            if (!PilotPassport::airfieldInPilotChallenge($airport->id, $flight->cid)) {
+                $this->info('- [Enrollment Limit] - airfield not part of an enrolled challenge');
+                continue;
+            }
+            if ($flight->altitude - $airport->elevation > SELF::ALTITUDE_LIMIT) {
+                $this->info('- [Altitude Limit] - too high log');
+                continue;
+            }
+            if (LatLong::calcDistance($ppos, $airport->fetchLatLong()) > SELF::RADIUS_LIMIT) {
+                $this->info('- [Radius Limit] - too far away to log');
+                continue;
+            }
+            return $airport->id;
+        }
+        return null;
+    }
+
+    public static function checkPhaseComplete(RealopsPilot $pilot) {
+        $pilot_has_visited = PilotPassportLog::where('cid', $pilot->id)->orderBy('airfield', 'asc')->pluck('airfield')->toArray();
+        $challenges = PilotPassport::get();
+        foreach ($challenges as $c) {
+            $challenge_airfields = PilotPassportAirfieldMap::where('mapped_to', $c->id)->orderBy('airfield', 'asc')->pluck('airfield')->toArray();
+            $challenge_complete = true;
+            foreach ($challenge_airfields as $c_a) {
+                if (!in_array($c_a, $pilot_has_visited)) {
+                    $challenge_complete = false;
+                    break;
+                }
+            }
+            if ($challenge_complete) {
+                $award = new PilotPassportAward;
+                $award->cid = $pilot->id;
+                $award->challenge_id = $c->id;
+                $award->awarded_on = date('Y-m-d H:i:s');
+                $award->save();
+            }
+            Mail::to($pilot->email)->send(new PilotPassportMail('phase_complete', $pilot, $c));
```

Comment:
I feel like this should be inside of the above if statement

Reply:
Done. Also added code to ensure that we aren't double-logging the phase complete.